src/errors.ts

```diff
@@ -82,8 +82,21 @@ export class MaxRelationSizeError extends Error {
   }
 }
 
+export class AbortError extends Error {
+  constructor(message?: string) {
+    super(message);
+
+    this.name = 'AbortError';
```

Comment:
sorry for jumping here, just noticed it. 
Why did you decide to extend it and put name? Is there any reason for that?
since as I know error name would be already 'AbortError' when it is aborted. Therefore, I think it does have to be extended. 

Here it is from documentation: 
https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort

> The reason why the operation was aborted, which can be any JavaScript value. If not specified, the reason is set to "AbortError"

Reply:
The main reason is that the error thrown through browser's fetch API is of type `DOMException` which is different than the one through the `node-fetch` API, which is explicitly defined as `AbortError`. Another thing, in the changes here, the abort is not passed to the `fetch` function, it's only used to break the polling loop and reject the promise manually.
It might be removed if we decided to pass the abort signal to the `fetch` API, but that requires further changes that should be discussed.